project/backend/src/routes/notes.js

```diff
@@ -0,0 +1,224 @@
+import express from "express"
+import { PrismaClient } from '../generated/prisma/index.js';
+
+const router = express.Router();
+const prisma = new PrismaClient();
+
+//Get all notes
+
+router.get("/", async (req, res)=> {
+    try{
+        const {
+          page = '1',
+          limit = '10',
+          search = '',
+          sortBy = 'createdAt',
+          sortOrder = 'desc',
+       } = req.query;
+
+       let where = {};
+       if (search) {
+           where = {
+               OR: [
+                { title: { contains: search, mode: 'insensitive' } },
+                { content: { contains: search, mode: 'insensitive' } },
+                ],
+           };
+        }
+
+        let orderDirection = 'desc'; // default
+        if (sortOrder === 'asc') {
+           orderDirection = 'asc';
+        }
+
+
+        const notes = await prisma.note.findMany({
+
+            //pagination
+            skip: (Number(page) - 1) * Number(limit),
+            take: Number(limit),
+
+            //search
+            where,
+
+            //sorting
+            orderBy: {
+               [sortBy]: orderDirection,
+            },
+        });
+
+        if(notes.length === 0 ){
+            res.status(200).json({message : "No notes found"});
+
+        } else {
+            res.status(200).json({
+              success: true,
+              data: notes
+            });
+        }
+        
+
+    } catch (error) {
+        console.error("Error fetching notes :" + error)
+        res.status(500).json({error : "Failed to fetch notes"})
+    }
+})

```

Comment:
@yasminebounasla you need to put always the controllers in a separated file for a better structure
```suggestion
router.get("/", getRoutes);
```
for example you could do something like that and import the getRoutes function from another file

Reply:
Noted thank u sm !